plexapi/utils.py

```diff
@@ -152,15 +152,18 @@ def threaded(callback, listargs):
             listargs (list): List of lists; \*args to pass each thread.
     """
     threads, results = [], []
+    connected_event = threading.Event()
     for args in listargs:
         args += [results, len(results)]
         results.append(None)
-        threads.append(Thread(target=callback, args=args))
+        threads.append(threading.Thread(target=callback, args=args, kwargs=dict(connected_event=connected_event)))
         threads[-1].setDaemon(True)
         threads[-1].start()
-    for thread in threads:
-        thread.join()
-    return results
+    while not connected_event.is_set():
```

Comment:
Just curious, what was the issue that led to this change?

Reply:
Most times I run the my things outside my home network, when the local IP of the server is not available, and it takes annoyingly long time to establish the connection

Reply:
Makes sense, thanks.